apollo-federation/src/sources/connect/expand/mod.rs

```diff
@@ -489,7 +488,8 @@ mod helpers {
                 //
                 // All sibling fields marked by $this in a transport must be carried over to the output type
                 // regardless of its use in the output selection.
-                let field_and_selection = FieldAndSelection::from_path(&path);
+                let joined = path.iter().map(|part| part.to_string()).join(".");
+                let field_and_selection = FieldAndSelection::from_path(&joined);
```

Comment:
Probably better to update `FieldAndSelection::from_path`, since it just splits this again. It's also only used right here.

Reply:
Nice, I'll make that change.